shop/src/main/java/ro/msg/learning/shop/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package ro.msg.learning.shop.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.domain.key.StockId;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.dto.StockDto;
+import ro.msg.learning.shop.exception.LocationNotFoundException;
+import ro.msg.learning.shop.exception.ProductCategoryNotFoundException;
+import ro.msg.learning.shop.exception.ProductNotFoundException;
+import ro.msg.learning.shop.mapper.ProductCategoryMapper;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.mapper.StockMapper;
+import ro.msg.learning.shop.message.Messages;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static ro.msg.learning.shop.message.Messages.*;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    @Autowired
```

Comment:
So there are many ways to do this DI in Spring and `@Autowired` is one of them. However there seems to be a better approach that we can use in order to achieve the same thing, but maybe more readable, testable and reliable.
This is nice to be implemented in projects that use lombok as we do.
Basically instead of using `@Autowired` we need to make sure that we have a constructor with all fields/arguments, and since this is the only constructor, Spring will inject the dependencies through it (the fields must be final).
References:

https://stackoverflow.com/questions/68538851/lombok-and-autowired
https://betulsahinn.medium.com/why-is-autowired-annotation-not-recommended-4939c46da1f8